kubernetes/k8s.yaml

```diff
@@ -1,52 +1,130 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: registry-rewriter-configmap
+  namespace: kube-system
+data:
+  default.conf: |-
+    server {
+            listen 443 ssl;
+
+            server_name localhost;
+            ssl_certificate /etc/nginx/ssl/tls.crt;
+            ssl_certificate_key /etc/nginx/ssl/tls.key;
+
+            location / {
+                    proxy_pass http://127.0.0.1:3000;
+            }
+    }
+---
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: local-repository-controller-deployment
+  name: registry-rewriter
+  namespace: kube-system
   labels:
-    app: local-repository-controller
+    app: registry-rewriter
+    mutateimagename: never
 spec:
-  replicas: 2
+  replicas: 1
   selector:
     matchLabels:
-      app: local-repository-controller
+      app: registry-rewriter
   template:
     metadata:
       labels:
-        app: local-repository-controller
+        app: registry-rewriter
     spec:
+      initContainers:
```

Comment:
I think this whole file can go in favour of using the helm chart to deploy the hook internally in a cluster.

Could make the chart so that the image name is configurable to something custom.